sdk/core/Azure.Core/generate_shared_ref_solution.ps1

```diff
@@ -0,0 +1,78 @@
+Set-StrictMode -Version Latest
+
+$RepoRoot = Resolve-Path $PSScriptRoot\..\..\..
+$ObjDirectory = "$RepoRoot\artifacts\obj";
+dotnet restore $RepoRoot\eng\service.proj
+
+Push-Location $PSScriptRoot
+
+$diff = git diff --name-only main -- $RepoRoot\sdk\core\Azure.Core\src\Shared\
+$sharedFiles = $diff -split "\n"
+For ($i=0; $i -lt $sharedFiles.Length; $i++) {
+    $sharedFiles[$i] = Split-Path $sharedFiles[$i] -Leaf
+}
+$sharedFilesPattern =  Join-String -Separator "|" -InputObject $sharedFiles
+$sharedFilesPattern = ("'" + $sharedFilesPattern + "'")
+
+try
+{
+    $slnName = "Azure.Core.SharedRef.All.sln";
+    if (!(Test-Path $slnName))
+    {
+        dotnet new sln -n "Azure.Core.SharedRef.All"
+    }
+    $dirs = Get-ChildItem -Directory $ObjDirectory
+    $projects = $dirs | %{ 
+        $assetsFile = "$_\project.assets.json"
+
+        if (!(Test-Path $assetsFile))
+        {
+            return;
+        }
+
+        Write-Host "Processing $assetsFile"
+        $assets = Get-Content -Raw $assetsFile;
+
+        if (($_.Name.StartsWith("Azure.")) -or ($assets -Match "Azure.Core"))
```

Comment:
Do we need this logic? can we look for .csproj references instead?

Reply:
I considered removing it, but I thought it would make sense to keep the same constraints as the original assets list. I'm fine with removing it if you think it could pick up additional references.

Reply:
I don't think you can reference core shared sources without referencing core.

Reply:
> Do we need this logic? can we look for .csproj containing shared source reference instead?

Also, the intent is to pick up not only direct references (which tend to be just the library itself) but test projects that indirectly use them through references to the library they reference. This is mostly to get tests for clients that reference shared source.